### PR TITLE
Common interface

### PR DESCRIPTION
**Pull request by mhasself**
Common framework for SAT and LAT noise models.  They're both in the main model file now, with very similar interfaces.

### CODE REVIEW

**Comment by mhasself** [comment]
Actually let me add the low-ell roll-off into this PR, too... I don't want to bump the version again...

File: so_models_v3/SO_Noise_Calculator_Public_v3_1_1.py [hunk 1]
```diff
@@ -0,0 +1,461 @@
...
+            A *= (60*180/np.pi)**2
+        elif units != 'sr':
+            raise ValueError("Unknown units '%s'." % units)
+        return A / self.get_survey_time()
+
+    def get_white_noise(self, f_sky, units='arcmin2'):
+        return self.band_sens**2 * self.get_survey_spread(f_sky, units=units)
+
+    def get_noise_curves(self, f_sky, ell_max, delta_ell, deconv_beam=True,
+                         full_covar=False):
```
Comment (zonca) [inline so_models_v3/SO_Noise_Calculator_Public_v3_1_1.py]: create docstring for the main interface for the user?

**Comment by mhasself** [comment]
@zonca I believe I have now duplicated all code as psuedo-code in the docstrings...

There's a roll-off function implemented here.  It turns out that smooth-transition-to-powerlaw is the best transfer function, and I set the default power to be strong enough to attenuate even the T noise in LAT.  It remains optional.  E.g. for cut-off at 200 the LAT TT result is:

image

I have no further commits coming unless there are further issues from @zonca or @msyriac .

File: so_models_v3/SO_Noise_Calculator_Public_v3_1_1.py [hunk 2]
```diff
@@ -0,0 +1,608 @@
...
+    approaching T(ell) \propto ell^-alpha.  The speed at which the
+    transition to the full power-law occurs is set by "patience";
+    patience is the maximum allowed value of:
+
+                       T(ell) * ell**alpha
+                 -----------------------------
+                  T(ell_off) * ell_off**alpha
+
+    I.e, supposing you were fighting an ell**alpha spectrum, how much
+    additional rise in power can you handle in thye transition?
```
Comment (zonca) [inline so_models_v3/SO_Noise_Calculator_Public_v3_1_1.py]: thye typo

File: so_models_v3/SO_Noise_Calculator_Public_v3_1_1.py [hunk 3]
```diff
@@ -0,0 +1,608 @@
...
+
+    to the maximum value is the factor by which an ell**alpha spectrum
+    is permitted to rise off quickly enough that an ell^alpha spectrum
+    times this transfer function will assymptotically approach
```
Comment (zonca) [inline so_models_v3/SO_Noise_Calculator_Public_v3_1_1.py]: assymptotically typo

File: so_models_v3/SO_Noise_Calculator_Public_v3_1_1.py [hunk 4]
```diff
@@ -0,0 +1,608 @@
...
+    patience is the maximum allowed value of:
+
+                       T(ell) * ell**alpha
+                 -----------------------------
+                  T(ell_off) * ell_off**alpha
+
+    I.e, supposing you were fighting an ell**alpha spectrum, how much
+    additional rise in power can you handle in thye transition?
+
+    to the maximum value is the factor by which an ell**alpha spectrum
```
Comment (zonca) [inline so_models_v3/SO_Noise_Calculator_Public_v3_1_1.py]: don't understand this sentence, is there maybe a typo?

**Comment by zonca** [comment]
It looks good to me, do we also want to change the ylim in the plot?

**Comment by zonca** [comment]
Also I think it would be useful to document here which value of ell_off and patience should be used in the sims.
so that if people are looking at these they know what to use if they want to reproduce what we have in the simulations

**Comment by mhasself** [comment]
I would prefer that the roll-off parameters be set by the calling entity (mapsims) rather than be a recommendation of the noise model.  I think most people agree that "at low ell, either the 1/f will kill us or some other property of the survey will mean we can't use those modes".  So the unrelenting rise of the default powerlaw sends a more accurate message than any rolled-off noise spectrum would.  We don't want people trying to push the experiment to low ell (higher f_sky), because their optimizer has concluded that the 1/f noise levels off eventually...

I think the roll-off should be used in application-specific ways that prevent an application from benefiting from increased signal/noise at the affected low ells.  One way to do this, as @msyriac has suggested, is to apply the transfer function to signal as well as to noise.  Equivalently, if you roll-off at some scale that will not be captured in your final extracted map footprints, that's fine too.

The sims will already require the right beams to be interpreted.  Data maps usually also have ell-space transfer functions (independent of the beam effects); so it's not crazy to have the sims include an ell-space transfer function, standing in for the map-maker transfer function, that pipelines should make use of.

W.r.t to plot limits: I only use the plots for comparison to older results... and so it's useful to leave them as is for the present change set.  Better limits could be a separate PR; no version bump required for that.

**Pull request merged**